pkg/helm/helm.go

```diff
@@ -308,14 +347,31 @@ func (h *Helm) LoadChart(ref *LoadChartConfig) (c *chart.Chart, err error) {
 	if err != nil {
 		return
 	}
+
 	log.Printf("Using chart path %v", chartPath)
+
 	// Check chart requirements to make sure all dependencies are present in /charts
-	if c, err = chartutil.Load(chartPath); err != nil {
+	if c, err = loadChartWithLockfile(chartPath, ref.LockFile); err != nil {
 		return
 	}
 
-	req, e := chartutil.LoadRequirements(c)
-	if e == nil {
+	lock, err := chartutil.LoadRequirementsLock(c)
+	if err != nil && err != chartutil.ErrLockfileNotFound {
+		return nil, err
+	}
+
+	req, err := chartutil.LoadRequirements(c)
+	if err != nil && err != chartutil.ErrRequirementsNotFound {
+		return nil, err
+	}
+
+	if req != nil {
+		if lock != nil {
+			if sum, err := resolver.HashReq(req); err != nil || sum != lock.Digest {
+				return nil, fmt.Errorf("requirements.lock is out of sync with requirements.yaml")
```

Comment:
@scjudd I keep hitting this error; the digest in my requirements.lock file was generated normally with `helm`.
How is a user supposed to compute the hash? Is it worth printing the expected hash as part of the error message?

Reply:
@james-callahan I guess your `requirements.lock` was generated using helm 3? This plugin still uses helm 2. Apparently a helm 3 `requirements.lock` is not compatible with helm 2. That can reproduced with helm directly:
```
$ helm3 dependency build example/localref/elk
# ...
$ helm2 dependency build example/localref/elk
Error: requirements.lock is out of sync with requirements.yaml
```
However a helm 2 `requirements.lock` can be used with helm 3 as well it seems.

You can obtain a proper `requirements.lock` using helm 2 or by rendering the chart without `requirements.lock` to let this plugin write the file.

At some point this plugin needs to support helm 3 as well. Unfortunately not all helm 2 charts are compatible with helm 3. Therefore after the helm 2 related changes here are done we could consider bumping this generator plugin's version to `v2` and use helm 3 - and eventually continue to maintain a separate `v1` branch for helm 2 support.

Reply:
@scjudd Nice work!
It would be great if you could add a test that verifies that the `requirements.lock` is handled properly.

Reply:
> @james-callahan I guess your `requirements.lock` was generated using helm 3? This plugin still uses helm 2. Apparently a helm 3 `requirements.lock` is not compatible with helm 2. 

I don't think so.
I just regenerated the requirements.lock file with helm 2 and its the same.


In helm2 I think they moved requirements.lock inside of Chart.yaml?

Reply:
I don't think that the requirements lock moved into the `Chart.yaml` file in helm 3.

Though I agree not having the plugin write the lockfile when it doesn't exist doesn't provide good usability.
The plugin should generate the lockfile itself (I imagine this would happen when `lockFile` is specified but the file does not exist).